### ISSUE TITLE
Incorrect casting of local functions to delegates results in errors about method groups

### ISSUE THREAD

**Issue by DavidArno** [dotnet/roslyn#14410]
**Version Used**: 
VS 15 Preview 5

**Steps to Reproduce**:
Create the following code in a C# file:

``` cs
static void Main()
{
    int LocalFunc(int x) => x * 2;
    var result = FuncWrapper(LocalFunc);
}

private static double FuncWrapper(Func<double, double> func) => func(2);
```

**Expected Behaviour**:
The compiler should report that it `cannot convert from local function 'localFunc' to 'Func<double, double>'`   

**Actual Behavior**:
The compiler complains that:
`Argument 1: cannot convert from 'method group' to 'Func<double, double>'`

**Comment by mburbea** [comment]
I get the same error for non-local functions. Are you sure a new error was added?
```csharp

        static int Foo(int x) => x * 2;

        static double Wrapper(Func<double, double> func) => func.Invoke(2);
        static void Main(string[] args)
        {
            var tryit = Wrapper(Foo);
        }
```

**Comment by gafter** [comment]
Not sure I understand. The error looks correct to me.

**Comment by DavidArno** [comment]
Hmm, I appear to have caused confusion for you both. I was arguing that the current choice of error message is misleading. I've no idea what you guys spec'ed for this feature, but as it stands, the error message seems wrong. 

If I define `LocalFunc` as a lambda, rather than a local function, I do not get a method group error. Even more oddly, if I define two local functions called `localFunc` in the one method, I get a "variable already exists" error. A local function can form closures, for example, over variables defined in the same method. It is semantically very different to a method group. The fact that it may well be turned into a method is an implementation detail and error messages should leak that implementation detail. This doesn't happen with lambdas, so why with local functions?

But if the message is not seen by the team as being wrong, then please close this as a "won't fix".

**Comment by agocke** [comment]
@DavidArno I think this is a bit of confusion about the C# spec. When you refer to a local function by name, the expression you receive is a "method group." This is the same as if you were to refer to a method by name. Lambdas, however, do not have names, so this is not a "method group conversion."

Since the distinction between a method group and a lambda is not implementation, but is actually part of the language specification, I think this error is correct as-is. If we want to provide a better error message for all types of method group conversions in the future I think that's worth considering, but I don't think local functions are special here.

I'm going to close this for now -- if you want to see a new error message for both methods and local functions, please open a new issue.
